### PR TITLE
Fix potential problems (StarGate)

### PR DESCRIPTION
**Pull request by zimzaza4**
## Changes

If the player's destination is not an alien world, don't give the player Metadata

## Related Issues

hmm
## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break world generation, mob spawning, and the like.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: src/main/java/io/github/addoncommunity/galactifun/base/items/StargateController.java [hunk 1]
```diff
@@ -328,7 +329,9 @@ public void onUsePortal(PlayerTeleportEndGatewayEvent e) {
 
         Block destBlock = b.getRelative(1, 0, 0);
         if (destBlock.getType().isEmpty()) {
-            e.getPlayer().setMetadata("CanTpAlienWorld", new FixedMetadataValue(Galactifun.instance(), true));
+            // Check if the destination is an alien world
+            AlienWorld world = Galactifun.worldManager().getAlienWorld(destBlock.getWorld());
+            if (world != null) e.getPlayer().setMetadata("CanTpAlienWorld", new FixedMetadataValue(Galactifun.instance(), true));
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/base/items/StargateController.java:334]: Add braces please

**Review by Seggan — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/addoncommunity/galactifun/base/items/StargateController.java (hunk 1 above)
Comment (zimzaza4) [reply src/main/java/io/github/addoncommunity/galactifun/base/items/StargateController.java:334]: ok

File: src/main/java/io/github/addoncommunity/galactifun/base/items/StargateController.java [hunk 2]
```diff
@@ -328,7 +329,11 @@ public void onUsePortal(PlayerTeleportEndGatewayEvent e) {
 
         Block destBlock = b.getRelative(1, 0, 0);
         if (destBlock.getType().isEmpty()) {
-            e.getPlayer().setMetadata("CanTpAlienWorld", new FixedMetadataValue(Galactifun.instance(), true));
+            // Check if the destination is an alien world
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/base/items/StargateController.java:332]: this comment is kinda misleading. maybe "check if the player is teleporting to an alien world, and if so, allow them to"?

**Review by Seggan — CHANGES_REQUESTED** [review]

**Review by Seggan — APPROVED** [review]

**Review by ProfElements — APPROVED** [review]
LGTM

**Pull request merged**